### ISSUE TITLE
Different score_threshold result in different mAP

### ISSUE THREAD

**Issue by yd8534976** [blue-oil/blueoil#1065]
In lmfyolo config file, we use default `score_threshold = 0.05`. If we change the `score_threshold` to another value. We may get different mAP.

Here, I try to change the `score_threshold` to 0. mAP will get larger. Results as below:

<img width="362" alt="Screen Shot 2020-05-28 at 16 46 31" src="https://user-images.githubusercontent.com/8168498/83113837-2111a600-a103-11ea-92ef-5501f1b43b9d.png">

<img width="1055" alt="Screen Shot 2020-05-28 at 16 46 10" src="https://user-images.githubusercontent.com/8168498/83113867-2cfd6800-a103-11ea-813d-a6d1a2abdcc0.png">

Maybe we should fix the `score_threshold` to a smaller value when training to calculate mAP.

**Comment by patrick-su** [comment]
@yd8534976 @masato0412
OK, after given some thoughts, I think this is what happened (assuming the mAP calculation is correct)
1. confidence score threshold lower
2. a lot of low-confidence boxes all count as prediction
3. since there is a lot of "random boxes", precision goes to 0%, and recall goes to 100%
4. however, for mAP, this only mean we have more sampling points on the right side of precisionxrecall curve, and have better approximation of real curve there.
5. area under curve (mAP) increase slightly due to better approximation

reference: https://github.com/rafaelpadilla/Object-Detection-Metrics
![prc](https://raw.githubusercontent.com/rafaelpadilla/Object-Detection-Metrics/master/aux_images/precision_recall_example_1_v2.png)

**Comment by joelN123** [comment]
yeah, it looks like there are some low score boxes that actually were correct? Maybe it is good to lower that threshold... Another point is that maybe the non-maximum suppression will have to do more work because there are more boxes to compare. Possibly it affects the speed of the post process.
